### ISSUE TITLE
.value is not resolved in template when use Nuxt plugin

### ISSUE THREAD

**Issue by misaon** [nuxt/framework#9541]
### Environment

- Operating System: `Linux`
- Node Version:     `v18.12.1`
- Nuxt Version:     `3.0.0`
- Nitro Version:    `1.0.0`
- Package Manager:  `pnpm@7.18.0`
- Builder:          `vite`
- User Config:      `modules`, `oxyshopNuxt`, `i18n`
- Runtime Modules:  `normalizedModule()`, `normalizedModule()`
- Build Modules:    `-`

### Reproduction

https://stackblitz.com/edit/github-xxv6jq?file=app.vue

### Describe the bug

If I define a `computed` with value in the **plugin**, then use the plugin in `<template>`, the `.value` does not contain any data. 
If I **inject** the plugin in `<script setup>` via `useNuxtApp`, computed is defined in the template.

**Comment by danielroe** [comment]
Indeed, Vue doesn't know to unwrap this value. It's tricky; even if we can handle this it would be a breaking change for existing APIs. I would advise instead injecting a reactive object from a plugin rather than an object containing refs (or, indeed, a ref).
